en/developer_guide/block_parameter.rst

```diff
@@ -0,0 +1,109 @@
+Block parameter
+===============
+
+Context
+-------
+
+To be properly displayed on the website, some block will require environment parameters.
+Those parameters could be of multiple type, for instance :
+
+ - url parameters
+ - server parameters
+ - header parameters
+ - ...
+
+An easy way to give those parameters to the block would have been to give all the request
+parameters when we call a block.
+
+This easy solution is not optimal a main reason, when working with `ESI`_, you will cache each block
+for each different url.
+
+``/block/1`` will be in one cache entry.
+``/block/1?foo=bar`` will be in another cache entry.
+
+Open Orchestra solution
+-----------------------
+
+At Open Orchestra, we prefer to give you the opportunity to pass only the parameters you want
+to request a block, this way, we would limit the cache entries for a single block.
+
+Backoffice work
+~~~~~~~~~~~~~~~
+
+In the Backoffice, we provide a way to store the parameter linked to the block in the block reference
+in the areas.
+
+You need to declare a service with the tag ``open_orchestra_backoffice.block_parameter.strategy``
+
+.. code-block:: yaml
+
+    tags:
+        - { name: open_orchestra_backoffice.block_parameter.strategy }
+
+The class should also implement the interface ``OpenOrchestra\Backoffice\BlockParameter\BlockParameterInterface``
+
+Frontoffice work
+~~~~~~~~~~~~~~~~
+
+In the Frontoffice, the request is created while you don't have any idea of the block type. You will have to create
+the request based only on the data stored in the ``blockParameter`` entry from the block reference.
+
+To analyse the parameters and complete the request, you need to declare a service with the tag ``open_orchestra_front.sub_query.strategy``
+
+.. code-block:: yaml
+
+    tags:
+        - { name: open_orchestra_front.sub_query.strategy }
+
+The class should also implement the interface ``OpenOrchestra\FrontBundle\SubQuery\SubQueryGeneratorInterface``
+
+Example
+-------
+
+Subquery strategies available
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
++------------------------------+------------------------------------------------------------+
+| Name                         | Action                                                     |
++==============================+============================================================+
+| ``DeviceSubQueryStrategy``   | Pass the header parameter ``x-ua-device``                  |
++------------------------------+------------------------------------------------------------+
+| ``PostDataSubQueryStrategy`` | Pass all data from the ``POST`` part                       |
++------------------------------+------------------------------------------------------------+
+| ``RequestSubQueryStrategy``  | Pass the specified request parameter (aliasId for instance |
++------------------------------+------------------------------------------------------------+
```

Comment:
In fact I'm not sure of how it works: can I have multiple strategies for a single block or is there a strategy per block ?